src/available-command.ts

```diff
@@ -30,6 +30,7 @@ export function generate(engine: Engine, subPhase: SubPhase = null, data?: any):
     case SubPhase.ChooseTechTile: return possibleTechTiles(engine, player);
     case SubPhase.CoverTechTile: return possibleCoverTechTiles(engine, player);
     case SubPhase.UpgradeResearch: return possibleResearchAreas(engine, player, "");
+    case SubPhase.UpgradeLowest: return possibleResearchAreas(engine, player, "", true);
```

Comment:
can be removed, and the line above can be changed to `case SubPhase.UpgradeResearch: return possibleResearchAreas(engine, player, !!data);`